src/julia.h

```diff
@@ -521,7 +521,6 @@ extern JL_DLLEXPORT jl_value_t *jl_stackovf_exception;
 extern JL_DLLEXPORT jl_value_t *jl_memory_exception;
 extern JL_DLLEXPORT jl_value_t *jl_readonlymemory_exception;
 extern JL_DLLEXPORT jl_value_t *jl_diverror_exception;
-extern JL_DLLEXPORT jl_value_t *jl_domain_exception;
 extern JL_DLLEXPORT jl_value_t *jl_overflow_exception;
 extern JL_DLLEXPORT jl_value_t *jl_inexact_exception;
```

Comment:
Speaking of that next PR, I'm noticing that the [elegant mechanism here](https://github.com/JuliaLang/julia/blob/063e8f110bb7f0ee61bbc166ad6d0af7e2113d87/base/boot.jl#L203-L204) doesn't seem to work as I'm suspecting it was intended:
```julia
be1(a, i) = (if i > 1 throw(BoundsError(a, i)) end; @inbounds ret = a[1]; ret)
be2(a, i) = (if i > 1 Base.throw_boundserror(a, i) end; @inbounds ret = a[1]; ret)
a = rand(5)

julia> @code_llvm be1(a, 2)

define double @julia_be1_62876(i8** dereferenceable(40), i64) #0 !dbg !5 {
top:
  %ptls_i8 = call i8* asm "movq %fs:0, $0;\0Aaddq $$-10928, $0", "=r,~{dirflag},~{fpsr},~{flags}"() #3
  %ptls = bitcast i8* %ptls_i8 to i8****
  %2 = alloca [3 x i8**], align 8
  %.sub = getelementptr inbounds [3 x i8**], [3 x i8**]* %2, i64 0, i64 0
  %3 = getelementptr [3 x i8**], [3 x i8**]* %2, i64 0, i64 2
  store i8** null, i8*** %3, align 8
  %4 = bitcast [3 x i8**]* %2 to i64*
  store i64 2, i64* %4, align 8
  %5 = getelementptr [3 x i8**], [3 x i8**]* %2, i64 0, i64 1
  %6 = bitcast i8* %ptls_i8 to i64*
  %7 = load i64, i64* %6, align 8
  %8 = bitcast i8*** %5 to i64*
  store i64 %7, i64* %8, align 8
  store i8*** %.sub, i8**** %ptls, align 8
  %9 = icmp slt i64 %1, 2
  br i1 %9, label %L5, label %if

if:                                               ; preds = %top
  %10 = call i8** @jlsys_Type_59491(i8** inttoptr (i64 140329698287600 to i8**), i8** nonnull %0, i64 %1)
  store i8** %10, i8*** %3, align 8
  call void @jl_throw(i8** %10)
  unreachable

L5:                                               ; preds = %top
  %11 = bitcast i8** %0 to double**
  %12 = load double*, double** %11, align 8
  %13 = load double, double* %12, align 8
  %14 = load i64, i64* %8, align 8
  store i64 %14, i64* %6, align 8
  ret double %13
}

julia> @code_llvm be2(a, 2)

define double @julia_be2_62879(i8** dereferenceable(40), i64) #0 !dbg !5 {
top:
  %2 = icmp slt i64 %1, 2
  br i1 %2, label %L5, label %if

if:                                               ; preds = %top
  call void @julia_throw_boundserror_62880(i8** nonnull %0, i64 %1)
  call void @llvm.trap()
  unreachable

L5:                                               ; preds = %top
  %3 = bitcast i8** %0 to double**
  %4 = load double*, double** %3, align 8
  %5 = load double, double* %4, align 8
  ret double %5
}
```

Obviously I can introduce `throw_inexacterror` if necessary.